### DISCUSSION TITLE
MQTT autodiscovery topics and measurement units in Acurite weather station

### DISCUSSION THREAD

**Discussion by shampeon** [merbanan/rtl_433#2782] [Q&A]
I just set up as Docker containers rtl_433 along with rtl_433_autodiscovery to wrap `rtl_433_mqtt_hass.py`. I can see the output of the sensors in my new Acurite 5-in-1 station now in Home Assistant:

<img width="331" alt="Screen Shot 2024-01-06 at 2 38 37 PM" src="https://github.com/merbanan/rtl_433/assets/5759834/f58336c9-0d4a-4cc7-b46f-7373ff022880">

I'm not totally sure why the sensors only have suffixes (e.g. `-F`, `-RT`, etc.).

I'm in the US, and have set the units to F/in/mph on the Acurite display, and the units are set to the same in Home Assistant. But as you can see the temperature is correctly outputting Fahrenheit, but the rain total and wind speed are in metric.

Looking through my autodiscovery topic in MQTT, I see that the sensor is converting from the imperial value back to metric for e.g. rain rate:

```json
{"name": "Acurite-5n1-B-2898-RT", "unit_of_measurement": "mm", "value_template": "{{ (float(value|float) * 25.4) | round(2) }}", "state_class": "total_increasing", "state_topic": "rtl_433/5687e923fbb8/devices/Acurite-5n1/B/2898/rain_in", "unique_id": "Acurite-5n1-B-2898-RT", "device": {"identifiers": ["Acurite-5n1-B-2898"], "name": "Acurite-5n1-B-2898", "model": "Acurite-5n1", "manufacturer": "rtl_433"}}
```

Note the state topic is `rain_in`. 

I'm a little confused why this conversion is happening. Setting `-C native` in rtl_433 doesn't have any effect.

What I want is for the sensor data in HA to match the global unit setting in HA.

In `rtl_433_mqtt_hass.py` these are the mappings for the sensors for `rain_mm` and `rain_in`:

```json
    "rain_mm": {
        "device_type": "sensor",
        "object_suffix": "RT",
        "config": {
            "device_class": "precipitation",
            "name": "Rain Total",
            "unit_of_measurement": "mm",
            "value_template": "{{ value|float|round(2) }}",
            "state_class": "total_increasing"
        }
    },
...
    "rain_in": {
        "device_type": "sensor",
        "object_suffix": "RT",
        "config": {
            "device_class": "precipitation",
            "name": "Rain Total",
            "unit_of_measurement": "mm",
            "value_template": "{{ (float(value|float) * 25.4) | round(2) }}",
            "state_class": "total_increasing"
        }
    },
```

Is it wrong that both `rain_mm` and `rain_in` have `"unit_of_measurement": "mm"`? Is that why the `-RT` sensor is showing in mm?

**Comment by zuckschwerdt** [comment]
That conversion snuck in with #1390 and likely should not be there. Units are selected with
`  [-C native | si | customary] Convert units in decoded output.`

**Comment by shampeon** [comment]
So changing my rtl433 container to use `-Ccustomary`:

~~~yaml
  rtl433:
    container_name: rtl433
    image: hertzg/rtl_433:latest
    devices:
      - "/dev/bus/usb/001/004"
    command:
      - "-Mtime:unix:usec:utc"
      - "-Mbits"
      - "-Mlevel"
      - "-Mprotocol"
      - "-Mstats:2:300"
      - "-Fmqtt://172.17.0.1:1883,user=xxxxxxx,pass=xxxxxxxx,retain=1"
      - "-Ccustomary"
~~~

Results in the correct units for everything _except_ `rain_in`.
<img width="332" alt="Screen Shot 2024-01-21 at 12 38 50 PM" src="https://github.com/merbanan/rtl_433/assets/5759834/973e9d07-17dc-47dc-a0b0-3e32a82d0987">

I'm thinking the fix is to flip the `value_template` for `rain_in` and `rain_mm`. E.g.:

~~~yaml
    "rain_mm": {
        "device_type": "sensor",
        "object_suffix": "RT",
        "config": {
            "device_class": "precipitation",
            "name": "Rain Total",
            "unit_of_measurement": "mm",
            "value_template": "{{ (float(value|float) * 25.4) | round(2) }}",
            "state_class": "total_increasing"
        }
    },
...
    "rain_in": {
        "device_type": "sensor",
        "object_suffix": "RT",
        "config": {
            "device_class": "precipitation",
            "name": "Rain Total",
            "unit_of_measurement": "in",
            "value_template": "{{ value|float|round(2) }}",
            "state_class": "total_increasing"
        }
    },
~~~

But what I'm not clear about is what happens when the device uses metric units.

**Reply by zuckschwerdt** [reply]
Your "rain_in" entry looks good, the "rain_mm" one should not be changed and is not used in your case, you always get rain_in messages.

**Comment by shampeon** [comment]
PR up: https://github.com/merbanan/rtl_433/pull/2801

**Comment by gdt** [comment]
With the PR being merged, is this discussion now about a fixed bug and should it be closed?  Or is there something else?

**Reply by shampeon** [reply]
The `rain_in` sensor now reads the correct values. But I've also noticed [`rain_rate_in_h`](https://github.com/merbanan/rtl_433/blob/master/examples/rtl_433_mqtt_hass.py#L443) has the incorrect `unit_of_measurement` and value template formula:

``` json
    "rain_rate_in_h": {
        "device_type": "sensor",
        "object_suffix": "RR",
        "config": {
            "device_class": "precipitation_intensity",
            "name": "Rain Rate",
            "unit_of_measurement": "mm/h",
            "value_template": "{{ (float(value|float) * 25.4) | round(2) }}",
            "state_class": "measurement"
        }
    },
```

**Reply by zuckschwerdt** [reply]
Fixed with 7f60fff
